app/assets/javascripts/common/reports.js

```diff
@@ -1693,15 +1695,25 @@ function baseBarChartConfig() {
 const dynamicChartSegementDashed = (
   ctx,
   numberOfXAxisTicks,
-  numberOfDashedSegments = 1
+  endDateKey,
+  monthsDashedBase = 1,
 ) => {
-  const dashStyle = [4, 3]
-  const segmentStartIndex = ctx.p0DataIndex
-  return isSegmentDashed(segmentStartIndex, numberOfXAxisTicks, numberOfDashedSegments) ? dashStyle : undefined;
+
+  const endDateKeyIndex = monthIndexFromDateKey(endDateKey);
+  const dateNowMonth = new Date().getMonth()
+
+  const monthsInPast = dateNowMonth - endDateKeyIndex;
+  const dashedSegments = Math.max(0, monthsDashedBase - monthsInPast);
+
+  const segmentStartNodeIndex = ctx.p0DataIndex
+  return isSegmentDashed(segmentStartNodeIndex, numberOfXAxisTicks, dashedSegments) 
+    ? [4, 3] // dashStyle
+    : undefined;
 };
 
-function isSegmentDashed(segmentStartIndex, numberOfXAxisTicks, segmentsToDashFromEnd) {
-  return segmentStartIndex >= numberOfXAxisTicks - (segmentsToDashFromEnd + 1)
+function isSegmentDashed(segmentStartNodeIndex, numberOfXAxisTicks, dashedSegments) {
+  const dataNodeToStartDashedLine = numberOfXAxisTicks - 1 - dashedSegments
+  return segmentStartNodeIndex >= dataNodeToStartDashedLine
```

Comment:
Could you please explain what is happening in this line? I am feeling little difficult to understand the logic here.

Reply:
```function isSegmentDashed(segmentStartNodeIndex, numberOfXAxisTicks, dashedSegments) {
  const dataNodeToStartDashedLine = numberOfXAxisTicks - 1 - dashedSegments
  return segmentStartNodeIndex >= dataNodeToStartDashedLine
};```
This function is run on every segment of the line - a segment is between 2 data points.
`dataNodeToStartDashedLine` is the index of the point (data node) where the dashed line begins. For example the dashed line is for 2 segments this number will be 15 (0 index).
`segmentStartNodeIndex` is actually `p0` for the segment but I wanted to be explicit in name - maybe this can be changed

when the `dataNodeToStartDashedLine` is equal to or greater than the current 'p0' the line is dashed